lib/msf/core/session_manager.rb

```diff
@@ -223,6 +223,10 @@ def register(session)
       end
     end
 
+    if session.respond_to?("on_registered")
```

Comment:
What are your thoughts on moving this into the `if session.register?` branch? 👀

Reply:
True - makes sense.